doc/pmreorder/pmreorder.1.md

```diff
@@ -0,0 +1,242 @@
+[comment]: <> ("AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT)
+[comment]: <> (LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR)
+[comment]: <> (A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT)
+[comment]: <> (OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,)
+[comment]: <> (SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT)
+[comment]: <> (LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,)
+[comment]: <> (DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY)
+[comment]: <> (THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT)
+[comment]: <> ((INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE)
+[comment]: <> (OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.)
+
+[comment]: <> (pmreorder.1 -- man page for daxio)
+
+[NAME](#name)<br />
+[SYNOPSIS](#synopsis)<br />
+[DESCRIPTION](#description)<br />
+[OPTIONS](#options)<br />
+[ENGINES](#engines)<br />
+[INSTRUMENTALIZATION](#instrumentalization)<br />
+[PMEMCHECK STORE LOG](#pmemcheck-store-log)<br />
+[EXAMPLE](#example)<br />
+[SEE ALSO](#see-also)<br />
+
+
+# NAME #
+
+**pmreorder** -- performs persistent consisteny check
+		 using store reordering mechanism
+
+
+# SYNOPSIS #
+
+```
+$ python pmreorder <options>
+```
+
+
+# DESCRIPTION #
+
+The pmreorder tool is a collection of python scripts designed
+to parse and replay operations logged by pmemcheck -
+a persistent memory checking tool.
+
+Pmreorder performs store reordering between prersistent
+memory barriers - a sequence of flush-fence operations.
+It uses the consistency checking routine provided in the
+command line options to check whether the files are in a
+consistent state.
+
+Seeing as the combination of valgrind and the multiple store
+reordering will take a lot of time to process, it would be
+beneficial to do as little stores as possible to verify
+whether the tested program is powerfail safe.
+
+
+# OPTIONS #
+
+`-h, --help`
+Prints synopsis and list of options.
+
+`-l <store_log>, --logfile <store_log>`
+The pmemcheck log file to process.
+
+`-c <prog|lib>, --checker <prog|lib>`
+Consistency checker type.
+
+`-p <path>, --path <path>`
+Path to the consistency checker.
+
+`-n <name>, --name <name>`
+The name of the consistency checking function
+within the library.
+
+`-t <print|file>, --output_type <print|file>`
+Set the logger type. Default value is `print`.
+
+`-o <pmreorder_output>, --output <pmreorder_output>`
+Set the logger output file.
+(If `-t` option is set to `file`).
+
+`-e <debug|info|warning|error|critical>,
+ --output_level <debug|info|warning|error|critical>`
+Set the output log level.
+
+`-r  <full|noreorder|accumulative|partial>,
+ --default_engine  <full|noreorder|accumulative|partial>`
+Set the default reorder engine. Default value is `full`.
+
+
+# ENGINES #
+
+By default all possible permutations of stores are tested.
```

Comment:
"By default, the `full` reorder engine is used, which means that for each set of stores, the tool generates and checks consistency of all possible store permutations.
This might prove to be very computationally expensive for most workloads. In order to facilitate tests that will finish in a reasonable time, but with reduced number of test sequences, a number of other engines was developed:"